python/ray/serve/backend_state_manager.py

```diff
@@ -977,12 +995,17 @@ def update(self) -> bool:
             slow_start_replicas = []
             for replica in replicas.pop(
                     states=[ReplicaState.STARTING_OR_UPDATING]):
-                ready, slow_start = replica.check_started()
+                (ready, slow_start,
+                 exception_in_constructor) = replica.check_started()
                 if ready:
                     # This replica should be now be added to handle's replica
                     # set.
                     replicas.add(ReplicaState.RUNNING, replica)
                     transitioned_backend_tags.add(backend_tag)
+                elif exception_in_constructor:
+                    # TODO(jiaodong): Probably better to handle it at replica level,
+                    # suggestions are very much welcomed :)
+                    self.delete_backend(backend_tag)
```

Comment:
Hmm, don't we also need to somehow set the exception in the goal manager here? Otherwise how will this get propagated to the future?